src/transformers/models/owlvit/modeling_owlvit.py

```diff
@@ -1304,15 +1392,110 @@ def image_text_embedder(
+
+    def image_image_embedder(
+        self,
+        query_pixel_values: torch.FloatTensor,
+        pixel_values: torch.FloatTensor,
+        output_attentions: Optional[bool] = None,
+        output_hidden_states: Optional[bool] = None,
+    ) -> Tuple[torch.FloatTensor]:
+        query_image_embeds = self.owlvit.get_image_features(
+            pixel_values=query_pixel_values,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_base_image_embeds=True,
+        )
+        image_embeds = self.owlvit.get_image_features(
+            pixel_values=pixel_values,
+            output_attentions=output_attentions,
+            output_hidden_states=output_hidden_states,
+            return_base_image_embeds=True,
+        )
+
+        # Normalize query image, image embeddings
+        query_image_embeds = query_image_embeds / query_image_embeds.norm(p=2, dim=-1, keepdim=True)
+        image_embeds = image_embeds / image_embeds.norm(p=2, dim=-1, keepdim=True)
+
+        # Resize class token
+        new_size = tuple(np.array(image_embeds.shape) - np.array((0, 1, 0)))
+        class_token_out = torch.broadcast_to(image_embeds[:, :1, :], new_size)
+
+        # Merge image embedding with class tokens
+        image_embeds = image_embeds[:, 1:, :] * class_token_out
+        image_embeds = self.layer_norm(image_embeds)
+
+        # Resize to [batch_size, num_patches, num_patches, hidden_size]
+        new_size = (
+            image_embeds.shape[0],
+            int(np.sqrt(image_embeds.shape[1])),
+            int(np.sqrt(image_embeds.shape[1])),
+            image_embeds.shape[-1],
+        )
+        image_embeds = image_embeds.reshape(new_size)
+
+        # Repeat the process for the query image
+        new_size = tuple(np.array(query_image_embeds.shape) - np.array((0, 1, 0)))
+        class_token_out = torch.broadcast_to(query_image_embeds[:, :1, :], new_size)
+
+        # Merge image embedding with class tokens
+        query_image_embeds = query_image_embeds[:, 1:, :] * class_token_out
+        query_image_embeds = self.layer_norm(query_image_embeds)
+
+        # Resize to [batch_size, num_patches, num_patches, hidden_size]
+        new_size = (
+            query_image_embeds.shape[0],
+            int(np.sqrt(query_image_embeds.shape[1])),
+            int(np.sqrt(query_image_embeds.shape[1])),
+            query_image_embeds.shape[-1],
+        )
+        query_image_embeds = query_image_embeds.reshape(new_size)
+
+        return (query_image_embeds, image_embeds)
+
     @add_start_docstrings_to_model_forward(OWLVIT_OBJECT_DETECTION_INPUTS_DOCSTRING)
     @replace_return_docstrings(output_type=OwlViTObjectDetectionOutput, config_class=OwlViTConfig)
     def forward(
         self,
-        input_ids: torch.Tensor,
         pixel_values: torch.FloatTensor,
+        input_ids: Optional[torch.Tensor] = None,
+        query_pixel_values: Optional[torch.FloatTensor] = None,
```

Comment:
We cannot accept this change, the first argument needs to stay the same otherwise every user's code that does not use keyword arguments only will break.

Reply:
@sgugger makes sense, I can create a new class / head for image-guided object detection.